### PR TITLE
MM-48246 - AB test open RHS with linked board

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds the code for running an AB-test for the user onboarding process where by default a board will be attached to the landing channel (Town-Square) and the user once he closes the onboarding list, the RHS panel will open automagically and point to the linked board.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48246

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/22201

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/onboarding_tasklist/onboarding_tasklist.tsx [hunk 1]
```diff
@@ -6,29 +6,37 @@ import {useDispatch, useSelector} from 'react-redux';
...
-import {FormattedMessage} from 'react-intl';
 
-import {useFirstAdminUser, useIsCurrentUserSystemAdmin} from 'components/global_header/hooks';
+import {FormattedMessage} from 'react-intl';
 
-import {getPrevTrialLicense} from 'mattermost-redux/actions/admin';
 import {getShowTaskListBool} from 'selectors/onboarding';
-import {getBool, getMyPreferences as getMyPreferencesSelector} from 'mattermost-redux/selectors/entities/preferences';
+import {shouldShowAutoLinkedBoard} from 'selectors/plugins';
+
```
Comment (pvev) [inline components/onboarding_tasklist/onboarding_tasklist.tsx:14]: most of these changes in the import section are just organizing the imports

File: components/search_results_header/search_results_header.tsx [hunk 2]
```diff
@@ -29,7 +29,6 @@ const BackButtonIcon = styled(LocalizedIcon)`
 
 type Props = {
     isExpanded: boolean;
-    channelId: string;
```
Comment (pvev) [inline components/search_results_header/search_results_header.tsx:32]: not related to this change, just removing unused param

File: components/search_results_header/search_results_header.test.tsx [hunk 3]
```diff
@@ -34,7 +33,6 @@ describe('search_results_header', () => {
     test('should NOT diplay expand when the parent is channel info', () => {
         renderWithIntl(
             <Header
-                channelId={'abcdef'}
```
Comment (pvev) [inline components/search_results_header/search_results_header.test.tsx:37]: not related to this change, just removing unused param

File: packages/components/src/tour_tip/tour_tip.scss [hunk 4]
```diff
@@ -1,3 +1,4 @@
+@import '../../../../sass/utils/mixins';
```
Comment (pvev) [inline packages/components/src/tour_tip/tour_tip.scss:1]: not sure if we want to pull webapp files this way into the package components. However I am not sure if there is a way to add upside paths to the tsconfig.
cc: @hmhealey @M-ZubairAhmed

File: plugins/rhs_plugin/index.js [hunk 5]
```diff
@@ -14,9 +18,14 @@ function mapStateToProps(state) {
     const pluginComponent = rhsPlugins.find((element) => element.id === pluggableId);
     const pluginTitle = pluginComponent ? pluginComponent.title : '';
 
+    const isBoards = pluginComponent.pluginId === suitePluginIds.focalboard || pluginComponent.pluginId === suitePluginIds.boards;
```
Comment (pvev) [inline plugins/rhs_plugin/index.js:21]: @hmhealey should we keep doing the validation for boards this way?

File: plugins/rhs_plugin/rhs_plugin.jsx [hunk 6]
```diff
@@ -25,6 +28,7 @@ export default class RhsPlugin extends React.PureComponent {
                 className='sidebar-right__body'
             >
                 <SearchResultsHeader>
+                    {this.props.showAutoLinkedBoardTourTip ? <AutoShowLinkedBoardTourTip/> : null}
```
Comment (pvev) [inline plugins/rhs_plugin/rhs_plugin.jsx:31]: @mhd-sln heads up this way I am adding the tourtip for the auto shown linked board

File: packages/components/src/tour_tip/tour_tip.scss (hunk 4 above)
Comment (hmhealey) [reply packages/components/src/tour_tip/tour_tip.scss:1]: That does seem like something we shouldn't be doing, but I don't know about Sass to know how it'll handle that, especially when we eventually get around to moving the web app code into a separate workspace. Doing this won't work for JS code, but maybe it'll work fine with CSS. Alternatively, we could consider moving these mixins into the components package and making them available for use outside of the package, but I don't know how that would be done in practice yet.

Anyway, I think that it's probably fine to do this if it works for now. At some point, we'll probably have to find a proper way to share these mixins between packages, but I don't think we need to worry about that yet

Comment (pvev) [reply packages/components/src/tour_tip/tour_tip.scss:1]: ok! thanks

File: actions/views/rhs.ts [hunk 7]
```diff
@@ -235,7 +235,6 @@ export function showChannelMembers(channelId: string, inEditingMode = false) {
 export function hideRHSPlugin(pluggableId: string) {
     return (dispatch: DispatchFunc, getState: GetStateFunc) => {
         const state = getState() as GlobalState;
-
```
Comment (mhd-sln) [inline actions/views/rhs.ts:238]: redundant file change

File: components/onboarding_tasklist/onboarding_tasklist.tsx [hunk 8]
```diff
@@ -183,11 +191,28 @@ const OnBoardingTaskList = (): JSX.Element | null => {
...
+    const findRhsPluginId = (pluginId: string) => {
+        const rhsPlugins = pluginsComponentsList.RightHandSidebarComponent;
+
+        if (rhsPlugins.length) {
+            for (const plugin of rhsPlugins) {
+                if (plugin.pluginId === pluginId) {
+                    return plugin.id;
+                }
+            }
+        }
```
Comment (mhd-sln) [inline components/onboarding_tasklist/onboarding_tasklist.tsx:212]: maybe something like
```suggestion
        return rhsPlugins.find(plugin => plugin.pluginId === pluginId)?.id;
```

File: plugins/rhs_plugin/index.js (hunk 5 above)
Comment (M-ZubairAhmed) [reply plugins/rhs_plugin/index.js:21]: Yes we still needs to check for both focalboard and boards

File: components/onboarding_tasklist/onboarding_tasklist.tsx (hunk 8 above)
Comment (pvev) [reply components/onboarding_tasklist/onboarding_tasklist.tsx]: nice suggestion! thanks!

**Review by julmondragon — APPROVED** [review]
Nice work @pvev !

**Review by stevemudie — APPROVED** [review]
Will test post merge.

**Review by mhd-sln — APPROVED** [review]
LGTM! Thanks @pvev!

**Pull request merged**